### ISSUE TITLE
"%La" & "%a" formats suffer "internal error" with some PPC long double (double-double) values

### ISSUE THREAD

**Issue by p5pRT** [Perl/perl5#14806]
Migrated from [rt.perl.org#125633](https://rt-archive.perl.org/perl5/Ticket/Display.html?id=125633) (status was 'resolved')

Searchable as RT125633$

**Comment by p5pRT** [comment]
### From @sisyphus

Hi\,

The double\-double format can encapsulate certain values up to 2098 bits of
precision\.
However\, if the double\-double encapsulates a value whose precision is
greater than 2093 bits\, attempting to print that value with "%a" or "%La"
formatting produces the error&#8203;:

Hexadecimal float&#8203;: internal error

So\, for example there's no problem with&#8203;:

$ perl \-le '$x = \(2\*\*1020\) \+ \(2\*\*\-1072\);printf "%La\\n"\, $x;'
0x1\.0000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000001p\+1020

But\, the error occurs with&#8203;:

$ perl \-le '$x = \(2\*\*1021\) \+ \(2\*\*\-1072\); printf "%La\\n"\, $x;'
Hexadecimal float&#8203;: internal error at \-e line 1\.

\(And it's the same when I use "%a" instead of "%La"\.\)

"perl \-V" output is&#8203;:

\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
Summary of my perl5 \(revision 5 version 22 subversion 0\) configuration&#8203;:

&nbsp;  Platform&#8203;:
&nbsp;    osname=linux\, osvers=3\.2\.0\-4\-powerpc64\,
archname=ppc64\-linux\-thread\-multi\-ld
&nbsp;    uname='linux debian\-sis 3\.2\.0\-4\-powerpc64 \#1 smp debian 3\.2\.68\-1\+deb7u1
ppc64 gnulinux '
&nbsp;    config\_args='\-des \-Duse64bitint \-Dusethreads \-Duselongdouble \-Duse64bitall
\-Dcc=gcc \-m64 \-Dprefix=/home/sisyphus\-sis/perl522\-m64\-multi\-ld'
&nbsp;    hint=recommended\, useposix=true\, d\_sigaction=define
&nbsp;    useithreads=define\, usemultiplicity=define
&nbsp;    use64bitint=define\, use64bitall=define\, uselongdouble=define
&nbsp;    usemymalloc=n\, bincompat5005=undef
&nbsp;  Compiler&#8203;:
&nbsp;    cc='gcc \-m64'\, ccflags
='\-D\_REENTRANT \-D\_GNU\_SOURCE \-fwrapv \-fno\-strict\-aliasing \-pipe \-fstack\-protector
\-I/usr/local/include \-D\_LARGEFILE\_SOURCE \-D\_FILE\_OFFSET\_BITS=64 \-D\_FORTIFY\_SOURCE=2'\,
&nbsp;    optimize='\-O1'\,
&nbsp;    cppflags='\-D\_REENTRANT \-D\_GNU\_SOURCE \-fwrapv \-fno\-strict\-aliasing \-pipe
\-fstack\-protector \-I/usr/local/include'
&nbsp;    ccversion=''\, gccversion='4\.6\.3'\, gccosandvers=''
&nbsp;    intsize=4\, longsize=8\, ptrsize=8\, doublesize=8\, byteorder=87654321\,
doublekind=4
&nbsp;    d\_longlong=define\, longlongsize=8\, d\_longdbl=define\, longdblsize=16\,
longdblkind=6
&nbsp;    ivtype='long'\, ivsize=8\, nvtype='long double'\, nvsize=16\, Off\_t='off\_t'\,
lseeksize=8
&nbsp;    alignbytes=16\, prototype=define
&nbsp;  Linker and Libraries&#8203;:
&nbsp;    ld='gcc \-m64'\, ldflags =' \-fstack\-protector \-L/usr/local/lib'
&nbsp;    libpth=/usr/local/lib /usr/lib/gcc/powerpc\-linux\-gnu/4\.6/include\-fixed
/usr/lib /lib/powerpc\-linux\-gnu /lib/\.\./lib /usr/lib/powerpc\-linux\-gnu
/usr/lib/\.\./lib /lib /lib64 /usr/lib64
&nbsp;    libs=\-lpthread \-lnsl \-ldl \-lm \-lcrypt \-lutil \-lc
&nbsp;    perllibs=\-lpthread \-lnsl \-ldl \-lm \-lcrypt \-lutil \-lc
&nbsp;    libc=libc\-2\.13\.so\, so=so\, useshrplib=false\, libperl=libperl\.a
&nbsp;    gnulibc\_version='2\.13'
&nbsp;  Dynamic Linking&#8203;:
&nbsp;    dlsrc=dl\_dlopen\.xs\, dlext=so\, d\_dlsymun=undef\, ccdlflags='\-Wl\,\-E'
&nbsp;    cccdlflags='\-fPIC'\,
lddlflags='\-shared \-O1 \-L/usr/local/lib \-fstack\-protector'


Characteristics of this binary \(from libperl\)&#8203;:
&nbsp;  Compile\-time options&#8203;: HAS\_TIMES MULTIPLICITY PERLIO\_LAYERS
&nbsp;                        PERL\_DONT\_CREATE\_GVSV
&nbsp;                        PERL\_HASH\_FUNC\_ONE\_AT\_A\_TIME\_HARD
&nbsp;                        PERL\_IMPLICIT\_CONTEXT PERL\_MALLOC\_WRAP
&nbsp;                        PERL\_NEW\_COPY\_ON\_WRITE PERL\_PRESERVE\_IVUV
&nbsp;                        USE\_64\_BIT\_ALL USE\_64\_BIT\_INT USE\_ITHREADS
&nbsp;                        USE\_LARGE\_FILES USE\_LOCALE USE\_LOCALE\_COLLATE
&nbsp;                        USE\_LOCALE\_CTYPE USE\_LOCALE\_NUMERIC USE\_LOCALE\_TIME
&nbsp;                        USE\_LONG\_DOUBLE USE\_PERLIO USE\_PERL\_ATOF
&nbsp;                        USE\_REENTRANT\_API
&nbsp;  Built under linux
&nbsp;  Compiled at Jun  2 2015 22&#8203;:14&#8203;:40
&nbsp;  @&#8203;INC&#8203;:
&nbsp;    /home/sisyphus\-sis/perl522\-m64\-multi\-ld/lib/site\_perl/5\.22\.0/ppc64\-linux\-thread\-multi\-ld
&nbsp;    /home/sisyphus\-sis/perl522\-m64\-multi\-ld/lib/site\_perl/5\.22\.0
&nbsp;    /home/sisyphus\-sis/perl522\-m64\-multi\-ld/lib/5\.22\.0/ppc64\-linux\-thread\-multi\-ld
&nbsp;    /home/sisyphus\-sis/perl522\-m64\-multi\-ld/lib/5\.22\.0
&nbsp;    \.
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#

Cheers\,
Rob

**Comment by p5pRT** [comment]
### From @sisyphus

Hi\,

The double\-double format can encapsulate certain values up to 2098 bits of
precision\.
However\, if the double\-double encapsulates a value whose precision is
greater than 2093 bits\, attempting to print that value with "%a" or "%La"
formatting produces the error&#8203;:

Hexadecimal float&#8203;: internal error

So\, for example there's no problem with&#8203;:

$ perl \-le '$x = \(2\*\*1020\) \+ \(2\*\*\-1072\);printf "%La\\n"\, $x;'
0x1\.0000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000001p\+1020

But\, the error occurs with&#8203;:

$ perl \-le '$x = \(2\*\*1021\) \+ \(2\*\*\-1072\); printf "%La\\n"\, $x;'
Hexadecimal float&#8203;: internal error at \-e line 1\.

\(And it's the same when I use "%a" instead of "%La"\.\)

"perl \-V" output is&#8203;:

\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
Summary of my perl5 \(revision 5 version 22 subversion 0\) configuration&#8203;:

&nbsp;  Platform&#8203;:
&nbsp;    osname=linux\, osvers=3\.2\.0\-4\-powerpc64\,
archname=ppc64\-linux\-thread\-multi\-ld
&nbsp;    uname='linux debian\-sis 3\.2\.0\-4\-powerpc64 \#1 smp debian 3\.2\.68\-1\+deb7u1
ppc64 gnulinux '
&nbsp;    config\_args='\-des \-Duse64bitint \-Dusethreads \-Duselongdouble \-Duse64bitall
\-Dcc=gcc \-m64 \-Dprefix=/home/sisyphus\-sis/perl522\-m64\-multi\-ld'
&nbsp;    hint=recommended\, useposix=true\, d\_sigaction=define
&nbsp;    useithreads=define\, usemultiplicity=define
&nbsp;    use64bitint=define\, use64bitall=define\, uselongdouble=define
&nbsp;    usemymalloc=n\, bincompat5005=undef
&nbsp;  Compiler&#8203;:
&nbsp;    cc='gcc \-m64'\, ccflags
='\-D\_REENTRANT \-D\_GNU\_SOURCE \-fwrapv \-fno\-strict\-aliasing \-pipe \-fstack\-protector
\-I/usr/local/include \-D\_LARGEFILE\_SOURCE \-D\_FILE\_OFFSET\_BITS=64 \-D\_FORTIFY\_SOURCE=2'\,
&nbsp;    optimize='\-O1'\,
&nbsp;    cppflags='\-D\_REENTRANT \-D\_GNU\_SOURCE \-fwrapv \-fno\-strict\-aliasing \-pipe
\-fstack\-protector \-I/usr/local/include'
&nbsp;    ccversion=''\, gccversion='4\.6\.3'\, gccosandvers=''
&nbsp;    intsize=4\, longsize=8\, ptrsize=8\, doublesize=8\, byteorder=87654321\,
doublekind=4
&nbsp;    d\_longlong=define\, longlongsize=8\, d\_longdbl=define\, longdblsize=16\,
longdblkind=6
&nbsp;    ivtype='long'\, ivsize=8\, nvtype='long double'\, nvsize=16\, Off\_t='off\_t'\,
lseeksize=8
&nbsp;    alignbytes=16\, prototype=define
&nbsp;  Linker and Libraries&#8203;:
&nbsp;    ld='gcc \-m64'\, ldflags =' \-fstack\-protector \-L/usr/local/lib'
&nbsp;    libpth=/usr/local/lib /usr/lib/gcc/powerpc\-linux\-gnu/4\.6/include\-fixed
/usr/lib /lib/powerpc\-linux\-gnu /lib/\.\./lib /usr/lib/powerpc\-linux\-gnu
/usr/lib/\.\./lib /lib /lib64 /usr/lib64
&nbsp;    libs=\-lpthread \-lnsl \-ldl \-lm \-lcrypt \-lutil \-lc
&nbsp;    perllibs=\-lpthread \-lnsl \-ldl \-lm \-lcrypt \-lutil \-lc
&nbsp;    libc=libc\-2\.13\.so\, so=so\, useshrplib=false\, libperl=libperl\.a
&nbsp;    gnulibc\_version='2\.13'
&nbsp;  Dynamic Linking&#8203;:
&nbsp;    dlsrc=dl\_dlopen\.xs\, dlext=so\, d\_dlsymun=undef\, ccdlflags='\-Wl\,\-E'
&nbsp;    cccdlflags='\-fPIC'\,
lddlflags='\-shared \-O1 \-L/usr/local/lib \-fstack\-protector'


Characteristics of this binary \(from libperl\)&#8203;:
&nbsp;  Compile\-time options&#8203;: HAS\_TIMES MULTIPLICITY PERLIO\_LAYERS
&nbsp;                        PERL\_DONT\_CREATE\_GVSV
&nbsp;                        PERL\_HASH\_FUNC\_ONE\_AT\_A\_TIME\_HARD
&nbsp;                        PERL\_IMPLICIT\_CONTEXT PERL\_MALLOC\_WRAP
&nbsp;                        PERL\_NEW\_COPY\_ON\_WRITE PERL\_PRESERVE\_IVUV
&nbsp;                        USE\_64\_BIT\_ALL USE\_64\_BIT\_INT USE\_ITHREADS
&nbsp;                        USE\_LARGE\_FILES USE\_LOCALE USE\_LOCALE\_COLLATE
&nbsp;                        USE\_LOCALE\_CTYPE USE\_LOCALE\_NUMERIC USE\_LOCALE\_TIME
&nbsp;                        USE\_LONG\_DOUBLE USE\_PERLIO USE\_PERL\_ATOF
&nbsp;                        USE\_REENTRANT\_API
&nbsp;  Built under linux
&nbsp;  Compiled at Jun  2 2015 22&#8203;:14&#8203;:40
&nbsp;  @&#8203;INC&#8203;:
&nbsp;    /home/sisyphus\-sis/perl522\-m64\-multi\-ld/lib/site\_perl/5\.22\.0/ppc64\-linux\-thread\-multi\-ld
&nbsp;    /home/sisyphus\-sis/perl522\-m64\-multi\-ld/lib/site\_perl/5\.22\.0
&nbsp;    /home/sisyphus\-sis/perl522\-m64\-multi\-ld/lib/5\.22\.0/ppc64\-linux\-thread\-multi\-ld
&nbsp;    /home/sisyphus\-sis/perl522\-m64\-multi\-ld/lib/5\.22\.0
&nbsp;    \.
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#

Cheers\,
Rob

**Comment by p5pRT** [comment]
### From @sisyphus



\-\-\-\-\-Original Message\-\-\-\-\- 
From&#8203;: Sisyphus \(via RT\)
Sent&#8203;: Friday\, July 17\, 2015 9&#8203;:49 PM
To&#8203;: bugs\-bitbucket@&#8203;rt\.perl\.org
Subject&#8203;: \[perl \#125632\] "%La" & "%a" formats suffer "internal error" with 
some PPC long double \(double\-double\) values

> The double\-double format can encapsulate certain values up to 2098 bits of 
> precision\.
> However\, if the double\-double encapsulates a value whose precision is 
> greater than 2093 bits\, attempting to print that value with "%a" or "%La" 
> formatting produces the error&#8203;:
>
> Hexadecimal float&#8203;: internal error

In sv\.c we have&#8203;:

if \(vend && \(vend \<= vhex || vend > vmaxend\)\)
&nbsp;    Perl\_croak\(aTHX\_ "Hexadecimal float&#8203;: internal error"\);

This is ok for up to 2093 bits
However\, for 2094 to 2097 bits vend > vmaxend by 1\, and perhaps \(untested\) 
greater by 2 for 2098 bits\.

Also\, in the same file\, we have&#8203;:

if \(v \<= vhex || v \- vhex >= VHEX\_SIZE ||
&nbsp;    /\* For double\-double the ixmin and ixmax stay at zero\,
&nbsp;     \* which is convenient since the HEXTRACTSIZE is tricky
&nbsp;     \* for double\-double\. \*/
&nbsp;    ixmin \< 0 || ixmax >= NVSIZE ||
&nbsp;    \(vend && v \!= vend\)\)
&nbsp;    Perl\_croak\(aTHX\_ "Hexadecimal float&#8203;: internal error"\);

The problem with this is that when we exceed 2093 bits of precision the 
condition \(v \- vhex >= VHEX\_SIZE\) becomes true\.
For 2094 to 2097 bits\, v\- vhex == VHEX\_SIZE \(525 ==525\)\.
And for 2098 bits\, v \-vhex > VHEX\_SIZE \(526 > 525\)

So \.\.\. for the former\, I changed the condition&#8203;:

vend > vmaxend
to
vend > vmaxend \+ 2

And\, for the latter\, I changed&#8203;:

v \- vhex >= VHEX\_SIZE
to
v \- vhex >= VHEX\_SIZE \+ 2

Perl\-5\.22\.0 rebuilt with those changes to its sv\.c seems to enable the 
correct %La output for the full range of values \- but then the following 
test scripts fail&#8203;:

op/pack\.t
porting/args\_assert\.t
porting/podcheck\.t

Incidentally\, is it right that "%a" and "%La" output identical strings ? 
\(That's what seems to happen for me\, and I've no objection to that \.\.\. but 
is it correct ?\)

That's all I've time for tonight\.

Cheers\,
Rob

**Comment by p5pRT** [comment]
### From @jhi

On Monday\-201507\-20 9&#8203;:55\, sisyphus1@&#8203;optusnet\.com\.au wrote&#8203;:
>
>
> \-\-\-\-\-Original Message\-\-\-\-\-
> From&#8203;: Sisyphus \(via RT\)
> Sent&#8203;: Friday\, July 17\, 2015 9&#8203;:49 PM
> To&#8203;: bugs\-bitbucket@&#8203;rt\.perl\.org
> Subject&#8203;: \[perl \#125632\] "%La" & "%a" formats suffer "internal error" with
> some PPC long double \(double\-double\) values
>
>> The double\-double format can encapsulate certain values up to 2098 bits of
>> precision\.
>> However\, if the double\-double encapsulates a value whose precision is
>> greater than 2093 bits\, attempting to print that value with "%a" or "%La"
>> formatting produces the error&#8203;:
>>
>> Hexadecimal float&#8203;: internal error
>
> In sv\.c we have&#8203;:
>
> if \(vend && \(vend \<= vhex || vend > vmaxend\)\)
>      Perl\_croak\(aTHX\_ "Hexadecimal float&#8203;: internal error"\);
>
> This is ok for up to 2093 bits
> However\, for 2094 to 2097 bits vend > vmaxend by 1\, and perhaps \(untested\)
> greater by 2 for 2098 bits\.
>
> Also\, in the same file\, we have&#8203;:
>
> if \(v \<= vhex || v \- vhex >= VHEX\_SIZE ||
>      /\* For double\-double the ixmin and ixmax stay at zero\,
>       \* which is convenient since the HEXTRACTSIZE is tricky
>       \* for double\-double\. \*/
>      ixmin \< 0 || ixmax >= NVSIZE ||
>      \(vend && v \!= vend\)\)
>      Perl\_croak\(aTHX\_ "Hexadecimal float&#8203;: internal error"\);
>
> The problem with this is that when we exceed 2093 bits of precision the
> condition \(v \- vhex >= VHEX\_SIZE\) becomes true\.
> For 2094 to 2097 bits\, v\- vhex == VHEX\_SIZE \(525 ==525\)\.
> And for 2098 bits\, v \-vhex > VHEX\_SIZE \(526 > 525\)
>
> So \.\.\. for the former\, I changed the condition&#8203;:
>
> vend > vmaxend
> to
> vend > vmaxend \+ 2
>
> And\, for the latter\, I changed&#8203;:
>
> v \- vhex >= VHEX\_SIZE
> to
> v \- vhex >= VHEX\_SIZE \+ 2

Can you try changing the \#define of VHEX\_SIZE instead \(for double\-double\)?

> Perl\-5\.22\.0 rebuilt with those changes to its sv\.c seems to enable the
> correct %La output for the full range of values \- but then the following
> test scripts fail&#8203;:
>
> op/pack\.t
> porting/args\_assert\.t
> porting/podcheck\.t
>
> Incidentally\, is it right that "%a" and "%La" output identical strings ?
> \(That's what seems to happen for me\, and I've no objection to that \.\.\. but
> is it correct ?\)

"Correct" is a big word\.\.\. at least it is consistent\.  The "L" in the 
original C\-land of course means "use long \(int\) instead of int"\,
or "long double" instead of "double"\.

As for Perl&#8203;: in a 32\-bit\-IV system\, with Perl 5\.10\.1\, using 'x' format 
instead&#8203;:

$ perl \-we 'printf "%x\\n"\, ~0'
ffffffff
$ perl \-we 'printf "%Lx\\n"\, ~0'
ffffffff

\(On a 64\-bit system\, ffffffffffffffff from both\.\)

In other words\, Perl doesn't really care about the 'L'\, it is more about 
the size of the IV\.

In general\, the C/printf worldview of "pull bytes \(as described by the 
format\) off the stack and generate a formatted string out of those 
bytes" is a little at odds with type\-safe\(r\) language like Perl\.

**Comment by p5pRT** [comment]
The RT System itself - Status changed from 'new' to 'open'

**Comment by p5pRT** [comment]
### From @sisyphus

\-\-\-\-\-Original Message\-\-\-\-\- 
From&#8203;: Jarkko Hietaniemi
Sent&#8203;: Wednesday\, July 22\, 2015 10&#8203;:25 PM
To&#8203;: sisyphus1@&#8203;optusnet\.com\.au ; perl5\-porters@&#8203;perl\.org ; 
bugs\-bitbucket@&#8203;rt\.perl\.org
Subject&#8203;: Re&#8203;: \[perl \#125632\] "%La" & "%a" formats suffer "internal error" 
with some PPC long double \(double\-double\) values

> Can you try changing the \#define of VHEX\_SIZE instead \(for double\-double\)?

I think that vmaxend \(which relates to HEXTRACTSIZE\, not VHEX\_SIZE\) is then 
still too small by 2\. So I still got the "internal error"\, but this time I 
also got the additional failing test scripts\.

I then instead tried setting DOUBLEDOUBLE\_MAXBITS to 3106 \- as another 
alternative way of increasing vmaxend and VHEX\_SIZE by the necessary 2\.
But\, although that eliminates the "internal error" and allows correct "%a" 
formatting\, porting/podcheck\.t and porting/args\_assert\.t still fail\. 
\(Outputs attached\.\)

I think op/pack\.t was failing anyway\.
I had either not noticed it failing when I originally built 5\.22\.0\, or had 
forgotten about it\. \(Perl\-5\.23\.1 is producing the same op/pack\.t failures\.\)

So they're not probably not relevant to this ticket\, but here are the 
op/pack\.t failures&#8203;:

\# Failed test 521 \- at op/pack\.t line 628
\# Failed test 694 \- at op/pack\.t line 628
\# Failed test 749 \- at op/pack\.t line 628
\# Failed test 804 \- at op/pack\.t line 628
\# Failed test 1024 \- at op/pack\.t line 628
\# Failed test 1079 \- at op/pack\.t line 628
\# Failed test 1134 \- at op/pack\.t line 628
\# Failed test 1354 \- at op/pack\.t line 628
\# Failed test 1409 \- at op/pack\.t line 628
\# Failed test 1464 \- at op/pack\.t line 628
\# Failed test 1684 \- at op/pack\.t line 628
\# Failed test 1739 \- at op/pack\.t line 628
\# Failed test 1794 \- at op/pack\.t line 628
\# Failed test 2014 \- at op/pack\.t line 628
\# Failed test 2069 \- at op/pack\.t line 628
\# Failed test 2124 \- at op/pack\.t line 628
\# Failed test 2344 \- at op/pack\.t line 628
\# Failed test 2399 \- at op/pack\.t line 628
\# Failed test 2454 \- at op/pack\.t line 628
\# Failed test 2894 \- at op/pack\.t line 628
\# Failed test 2949 \- at op/pack\.t line 628
\# Failed test 3004 \- at op/pack\.t line 628
\# Failed test 3059 \- at op/pack\.t line 628
\# Failed test 3444 \- at op/pack\.t line 628
\# Failed test 3499 \- at op/pack\.t line 628
\# Failed test 13238 \- at op/pack\.t line 628
\# Failed test 13293 \- at op/pack\.t line 628
\# Failed test 13348 \- at op/pack\.t line 628

Cheers\,
Rob

**Comment by p5pRT** [comment]
### From @sisyphus

[args_assert.fail.gz](https://rt-archive.perl.org/perl5/Ticket/Attachment/1358152/726632/args_assert.fail.gz)

**Comment by p5pRT** [comment]
### From @sisyphus

[podcheck.fail.gz](https://rt-archive.perl.org/perl5/Ticket/Attachment/1358152/726633/podcheck.fail.gz)

**Comment by p5pRT** [comment]
### From @jhi

> So they're not probably not relevant to this ticket\, but here are the 
> op/pack\.t failures&#8203;:
> 
> \# Failed test 521 \- at op/pack\.t line 628
> \# Failed test 694 \- at op/pack\.t line 628
\.\.\.

Ack\.  The pack\(\) code is separate\(\) from the printf\(\) code\.  Can you file another bug on that?

**Comment by p5pRT** [comment]
### From @jhi

> porting/podcheck\.t and porting/args\_assert\.t still fail

These failures look utterly unrelated to anything to do with floating point\.  They look more like a full reset \("git clean \-dxf"\, if using git\, "rm \-f config\.sh Policy\.sh config\.arch" if not\) and re\-Configure\-etc is needed\.

**Comment by p5pRT** [comment]
### From @jhi

Could you try the attached patch?
\[Scratch that\.  For some reason Chrome's file browser has decided that my patch file doesn't exist\.\.\.\]
I'll email it to you directly\.

**Comment by p5pRT** [comment]
### From @jhi

The 5488d37 should fix this problem\.  Please verify\.

**Comment by p5pRT** [comment]
### From [Unknown Contact. See original ticket]

The 5488d37 should fix this problem\.  Please verify\.

**Comment by p5pRT** [comment]
### From @tonycoz

On Thu Aug 06 15&#8203;:47&#8203;:33 2015\, jhi wrote&#8203;:
> The 5488d37 should fix this problem\.  Please verify\.

I've fixed it\, but the patch also introduced a UTF\-8 encoded BOM\, which broke
the build on NetBSD\.

Tony

**Comment by p5pRT** [comment]
### From @jhi

Wow\. A real UTF\-8 encoded BOM\. Thanks for fixing that\, I should have
noticed that\.

I wonder why NetBSD\.\.\. old enough/too old gcc?

I would suggest a porting test \(for\, say\, top level \*\.\[hc\]\) if I had not
the feeling that we have enough stuff there already\.\.\.

On Friday\, August 7\, 2015\, Tony Cook via RT \<perlbug\-followup@&#8203;perl\.org>
wrote&#8203;:

> On Thu Aug 06 15&#8203;:47&#8203;:33 2015\, jhi wrote&#8203;:
> > The 5488d37 should fix this problem\.  Please verify\.
>
> I've fixed it\, but the patch also introduced a UTF\-8 encoded BOM\, which
> broke
> the build on NetBSD\.
>
> Tony
>
>

\-\- 
There is this special biologist word we use for 'stable'\. It is 'dead'\. \-\-
Jack Cohen

**Comment by p5pRT** [comment]
### From @sisyphus

\-\-\-\-\-Original Message\-\-\-\-\- 
From&#8203;: Jarkko Hietaniemi via RT
Sent&#8203;: Friday\, August 07\, 2015 8&#8203;:47 AM
To&#8203;: OtherRecipients of perl Ticket \#125633&#8203;:
Cc&#8203;: perl5\-porters@&#8203;perl\.org
Subject&#8203;: \[perl \#125633\] "%La" & "%a" formats suffer "internal error" with 
some PPC long double \(double\-double\) values

> The 5488d37 should fix this problem\.  Please verify\.

Current blead \(up to commit 749f0eb\) looks fine as regards this issue\.
I think this ticket \(along with the duplicate \#126632\) can be closed\.

Thank you\.

Cheers\,
Rob

**Comment by p5pRT** [comment]
### From @sisyphus

>> The 5488d37 should fix this problem\.  Please verify\.
>
> Current blead \(up to commit 749f0eb\) looks fine as regards this issue\.
> I think this ticket \(along with the duplicate \#126632\) can be closed\.

Duh \.\.\. it is\, of course\, \#125632 that's the dup\.

Rob\.

**Comment by p5pRT** [comment]
### From @jhi

Sisyphus confirmed the fix\, resolving\.

**Comment by p5pRT** [comment]
@jhi - Status changed from 'open' to 'resolved'
